lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -54,67 +55,98 @@ export class Vulnerabilities {
     config: RenovateConfig,
     packageFiles: Record<string, PackageFile[]>
   ): Promise<void> {
+    const vulnerabilityGroups = await this.fetchVulnerabilityGroups(
+      config,
+      packageFiles
+    );
+
+    config.packageRules ??= [];
+    for (const { vulnerabilities, versioningApi } of vulnerabilityGroups) {
+      const groupPackageRules: PackageRule[] = [];
+      for (const vulnerability of vulnerabilities) {
+        const rule = this.vulnerabilityToPackageRules(vulnerability);
+        if (is.nullOrUndefined(rule)) {
+          continue;
+        }
+        groupPackageRules.push(rule);
+      }
+      this.sortByFixedVersion(groupPackageRules, versioningApi);
+
+      config.packageRules.push(...groupPackageRules);
+    }
+  }
+
+  async fetchVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>
+  ): Promise<Vulnerability[]> {
+    const groups = await this.fetchVulnerabilityGroups(config, packageFiles);
+    return groups.flatMap((group) => group.vulnerabilities);
+  }
+
+  async fetchVulnerabilityGroups(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>
+  ): Promise<VulnerabilityGroup[]> {
     const managers = Object.keys(packageFiles);
     const allManagerJobs = managers.map((manager) =>
       this.fetchManagerVulnerabilities(config, packageFiles, manager)
     );
-    await Promise.all(allManagerJobs);
+    return (await Promise.all(allManagerJobs)).flat();
   }
 
   private async fetchManagerVulnerabilities(
     config: RenovateConfig,
     packageFiles: Record<string, PackageFile[]>,
     manager: string
-  ): Promise<void> {
+  ): Promise<VulnerabilityGroup[]> {
     const managerConfig = getManagerConfig(config, manager);
     const queue = packageFiles[manager].map(
-      (pFile) => (): Promise<void> =>
-        this.fetchManagerPackageFileVulnerabilities(
-          config,
-          managerConfig,
-          pFile
-        )
+      (pFile) => (): Promise<VulnerabilityGroup[]> =>
+        this.fetchManagerPackageFileVulnerabilities(managerConfig, pFile)
     );
     logger.trace(
       { manager, queueLength: queue.length },
       'fetchManagerUpdates starting'
```

Comment:
Not related to this PR but just spotted: `fetchManagerUpdates` should actually be `fetchManagerVulnerabilities` (also 3 lines below)

Reply:
```suggestion
      'fetchManagerVulnerabilities starting'
```